cranelift/assembler-x64/meta/src/generate/format.rs

```diff
@@ -301,6 +334,104 @@ impl dsl::Format {
         style
     }
 
+    fn generate_evex_prefix(&self, f: &mut Formatter, evex: &dsl::Evex) -> ModRmStyle {
+        use dsl::OperandKind::{FixedReg, Imm, Mem, Reg, RegMem};
+
+        f.empty_line();
+        f.comment("Emit EVEX prefix.");
+        let (l_bit, l_prime_bit) = evex.length.bits();
+        fmtln!(f, "let ll = ({:#02b}, {:#02b});", l_bit, l_prime_bit);
+        fmtln!(f, "let pp = {:#04b};", evex.pp.map_or(0b00, |pp| pp.bits()));
+        fmtln!(f, "let mmmmm = {:#07b};", evex.mmmmm.unwrap().bits());
+        fmtln!(f, "let w = {};", evex.w.as_bool());
+
+        let bits = if let Some(aaa_bit) = self.mask_register() {
+            fmtln!(f, "let k_reg = {};", aaa_bit);
+            fmtln!(f, "let zeroing = {};", self.zeroing());
+            // Broadcast bit set to false till implemented.
+            fmtln!(f, "let b = false;");
+            format!("ll, pp, mmmmm, w, b, Some((k_reg, zeroing))")
+        } else {
+            format!("ll, pp, mmmmm, w, b, None")
+        };
+
+        let style = match self.operands_by_kind().as_slice() {
+            [Reg(reg), Reg(vvvv), Reg(rm)] => {
+                fmtln!(f, "let reg = self.{reg}.enc();");
+                fmtln!(f, "let vvvv = self.{vvvv}.enc();");
+                fmtln!(f, "let rm = self.{rm}.encode_bx_regs();");
+                fmtln!(f, "let evex = EvexPrefix::new(reg, vvvv, rm, {bits});");
+                ModRmStyle::Reg {
+                    reg: ModRmReg::Reg(*reg),
+                    rm: *rm,
+                }
+            }
+            [Reg(reg), Reg(vvvv), RegMem(rm)]
+            | [Reg(reg), Reg(vvvv), Mem(rm)]
+            | [Reg(reg), Reg(vvvv), RegMem(rm), Imm(_) | FixedReg(_)]
+            | [Reg(reg), RegMem(rm), Reg(vvvv)] => {
+                fmtln!(f, "let reg = self.{reg}.enc();");
+                fmtln!(f, "let vvvv = self.{vvvv}.enc();");
+                fmtln!(f, "let rm = self.{rm}.encode_bx_regs();");
+                fmtln!(f, "let evex = EvexPrefix::new(reg, vvvv, rm, {bits});");
+                ModRmStyle::RegMem {
+                    reg: ModRmReg::Reg(*reg),
+                    rm: *rm,
+                }
+            }
+            [Reg(reg), Reg(vvvv), RegMem(rm), Reg(is4)] => {
```

Comment:
I'm not sure that we want to copy-paste these yet from the VEX implementation. I suspect they all are the same but I'm not really sure, so we might want to start with only the formats we need and build from there. At some later point we could even deduplicate between VEX and EVEX.